packages/ui/src/CodeEditor.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+  CodeMirror based Code Editor component.
+
+  * For readonly/code-viewing purposes use `Code`.
+
+  ## CM5 vs CM6
+
+  We are not using the old CM5; we're basing our component on CM6 which is the next gen. version of the hugely
+  popular CM5. CM6 is still years old and solid despite being a young project.
+
+  See https://codemirror.net/6/ for more.
+
+  ## About language support
+
+  In CM6 all of the languages are defined in small, self defined packages.
+  ie. import { javascript } from '@codemirror/lang-javascript'
+
+  But it also supports a legacy mode for which there is a combined package of
+  lots of languages (coming from CM5 - which was hugely popular for a decade).
+
+  We are using this legacy language pack for ease of use.
+
+  So using a language is one import and one property assignment. See the `CodeEditor.stories.tsx`.
+
+  for more see https://github.com/codemirror/legacy-modes
+
+  ## About styling/themes
+
+  CM6 is not that rich with predefined themes yet so we are not providing anything other than the
+  default theme (though it's easy provide custom ones if/when we need it.)
+
+  ## Customizing the editor
```

Comment:
Should we customize the default theme with our colors and our fonts?

Reply:
another case of "do we want that?". in MC (sql-browser) we're using one of the default themes (solarized). i could backport that if you think we should.

Reply:
@utkukaratas I would like the code editor to use our new fonts and colors, not the old ones. Just for it to look consistent with the rest of the UI. If you are in doubt, I would suggest consulting with @pawelklasa. However, we might want to prototype something first ourselves not to be blocked by him.

Reply:
luckily the default theme is inline with our new MC colors with the white/blue/grays. let me show you and pawel things live when redesign work is done this week and we could backport any theme if we don't like it. resolving this.

Reply:
Are they *our* gray, blue and green though? What about the font?